concrete/concrete.go

```diff
@@ -100,34 +100,30 @@ func NewRegistry() *GenericPrecompileRegistry {
 }
 
 func (c *GenericPrecompileRegistry) index(blockNumber uint64) int {
-	for ii, startingBlock := range c.startingBlocks {
+	// Iterate over blocks and return index if found else return -1
+	for idx, startingBlock := range c.startingBlocks {
 		if blockNumber < startingBlock {
-			continue
-		}
-		if ii == len(c.startingBlocks)-1 {
-			return ii
-		}
-		if blockNumber < c.startingBlocks[ii+1] {
-			return ii
+			return idx - 1
 		}
 	}
-	return -1
+	return len(c.startingBlocks) - 1
 }
 
 func (c *GenericPrecompileRegistry) AddPrecompiles(startingBlock uint64, precompiles PrecompileMap) {
 	idx := c.index(startingBlock)
 	if idx >= 0 && c.startingBlocks[idx] == startingBlock {
 		panic("precompiles already set for this block")
-	}
+	} else {
+		//Append the addresses and then insert
```

Comment:
These insert statements should not be in an else block, the ones in `AddPrecompile` should. We are checking for an unexpected condition that causes a panic, not handling two expected possibilities.

Reply:
Got it will make necessary changes